core/ir/x86/instr.c

```diff
@@ -1346,21 +1346,18 @@ instr_may_write_zmm_or_opmask_register(instr_t *instr)
 {
     if (instr_get_prefix_flag(instr, PREFIX_EVEX))
         return true;
-    if (instr_raw_bits_valid(instr)) {
+    if (instr_raw_bits_valid(instr) &&
```

Comment:
I'm curious about this check: is it that it is significantly faster than the one below? It's certainly not complete, as it misses out on `kmovw`

Reply:
If it's broken maybe just remove it.  For plain DR with no client this would be faster as then there is no decoding required beyond level 1 (just find the size) which is all DR does for non-"interesting" instructions -- but OTOH does core DR care about zmm-writing?  Is this function called by core DR at all?

Reply:
Yeah, this is invoked in `client_process_bb`; DR tries to figure out whether there's some AVX instr at play. And if yes, it enables saving/restoring zmm and opmasks.

Reply:
I'll still need to check if the raw bits check is just an optimisation or if it's functionally different.

Reply:
But `client_process_bb` is only for when a client is present, for which `full_decode` will be set and so this optimization will not be needed.

Reply:
I see. `full_decode` is set here if there are clients: https://github.com/DynamoRIO/dynamorio/blob/c6474fda6ad45c88944dbadd79f870ade46c7ee2/core/arch/interp.c#L5035. So decoded instr is available and can be used directly, instead of the brittle (and slightly complex) bytes check. I'll remove the bytes check.